src/components/common/popup/index.tsx

```diff
@@ -0,0 +1,34 @@
+import { useAppSelector } from "hooks";
+import { popupSelector, PopupType } from "store/reducers/popupReducer";
+import CategoryForm, { CategoryFormProps } from "./CategoryForm";
+import DeleteConfirm, { DeleteConfirmProps } from "./DeleteConfirm";
+import ItemForm, { ItemFormProps } from "./ItemForm";
+import LoginConfirm, { LoginConfirmProps } from "./LoginConfirm";
+import NotificationMessage, { NotificationMessageProps } from "./NotificationMessage";
+
+type PopupMapType = {
+  [PopupType.CATEGORY_FORM]: React.FC<CategoryFormProps>;
+  [PopupType.ITEM_FORM]: React.FC<ItemFormProps>;
+  [PopupType.LOGIN_CONFIRM]: React.FC<LoginConfirmProps>;
+  [PopupType.DELETE_CONFIRM]: React.FC<DeleteConfirmProps>;
+  [PopupType.NOTIFICATION_MESSAGE]: React.FC<NotificationMessageProps>;
+};
```

Comment:
Type should be generic.

```suggestion
type PopupMapType = Record<PopupType, ...>;
```